### PR TITLE
Feature 1451/create permission endpoint

### PR DESCRIPTION
**Pull request by HootDunk**

### CODE REVIEW

**Review by moshirim — APPROVED** [review]

File: server/src/test/java/com/objectcomputing/checkins/services/permissions/PermissionsControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        assertEquals(HttpStatus.OK, response.getStatus());
+        assertTrue(response.getBody().isPresent());
+        assertEquals(response.getBody().get(), list);
+    }
+
+    @Test
+    void testGetAllPermissionsEnsureAlphabeticalOrder() {
+        Permission lastPermission = createACustomPermission("z");
+        Permission firstPermission = createACustomPermission("A");
+        Permission middlePermisison = createACustomPermission("H");
```
Comment (Bagurp) [inline server/src/test/java/com/objectcomputing/checkins/services/permissions/PermissionsControllerTest.java:54]: Typo here

File: server/src/test/java/com/objectcomputing/checkins/services/permissions/PermissionsControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    void testGetPermissionsNoneExistsReturnsEmptyList(){
+        final HttpRequest<Object> request = HttpRequest.
+                GET("/").basicAuth(MEMBER_ROLE, MEMBER_ROLE);
+
+        final HttpResponse<List<Permission>> response =
+                client.toBlocking().exchange(request, Argument.listOf(Permission.class));
+
+        assertEquals(HttpStatus.OK, response.getStatus());
+        Assertions.assertTrue(response.getBody().isPresent());
+        assertEquals(response.getBody().get(), List.of());
```
Comment (Bagurp) [inline server/src/test/java/com/objectcomputing/checkins/services/permissions/PermissionsControllerTest.java:77]: maybe assertTrue(response.getBody().get().isEmpty())?

**Review by Bagurp — APPROVED** [review]

**Comment by Bagurp** [comment]
@HootDunk, looks like adding Swagger documentation is part of AC. Will swagger documentation be automatically updated or should we manually write it for the endpoints?

File: server/src/test/java/com/objectcomputing/checkins/services/permissions/PermissionsControllerTest.java (hunk 2 above)
Comment (HootDunk) [reply server/src/test/java/com/objectcomputing/checkins/services/permissions/PermissionsControllerTest.java:77]: Yeah that's a better way doing that, will update.

**Comment by HootDunk** [comment]
> @HootDunk, looks like adding Swagger documentation is part of AC. Will swagger documentation be automatically updated or should we manually write it for the endpoints?

It gets automatically generated as long as the right comments and annotations are in place.  For controllers, you place an @Tag annotation at the class level which groups the methods together under whatever name was given as the value for the @Tag.   Then the comments above each controller method are how swagger is able to generate the documentation for each method.  For entities you simply add an @Schema annotation over each field to describe it's purpose and it will be included in swagger with the other entities on the bottom.

**Pull request merged**